bokehjs/src/coffee/core/util/types.ts

```diff
@@ -0,0 +1,30 @@
+const toString = Object.prototype.toString
+
+export function isBoolean(obj: any): obj is boolean {
+  return obj === true || obj === false || toString.call(obj) === '[object Boolean]'
+}
+
+export function isNumber(obj: any): obj is number {
+  return toString.call(obj) === "[object Number]"
+}
+
+export function isString(obj: any): obj is string {
+  return toString.call(obj) === "[object String]"
+}
+
+export function isStrictNaN(obj: any): obj is number {
+  return isNumber(obj) && obj !== +obj
+}
```

Comment:
We use both `global.isNaN()` and this function, which have fairly different corner case behavior. I didn't want to investigate all use cases of `global.isNaN()` right now, so I proceeded this way, so that there is no room for confusion which is which. In a later iteration I will have to resolve this, because ultimately there should be only one `isNaN()`.